### PR TITLE
feat: Add `install` command to pre-install required plugins

### PR DESCRIPTION
**Pull request by disq**
Fixes https://github.com/cloudquery/cloudquery/issues/13937
Requires https://github.com/cloudquery/plugin-pb-go/pull/103 (merged/tagged)

### CODE REVIEW

File: cli/cmd/root.go [hunk 1]
```diff
@@ -127,6 +127,7 @@ func NewCmdRoot() *cobra.Command {
 		NewCmdSync(),
 		NewCmdMigrate(),
 		newCmdDoc(),
+		newCmdDownload(),
```
Comment (disq) [inline cli/cmd/root.go:130]: Export or not export?

File: cli/go.mod [hunk 2]
```diff
@@ -68,3 +68,5 @@ require (
 )
 
 replace github.com/apache/arrow/go/v14 => github.com/cloudquery/arrow/go/v14 v14.0.0-20230916001126-ffb7089245ac
+
+replace github.com/cloudquery/plugin-pb-go => github.com/cloudquery/plugin-pb-go v1.9.5-0.20230919081227-761ada95bcf0
```
Comment (disq) [inline cli/go.mod:72]: Using untagged version for now

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
This looks good 🚀  Just missing a test for the new command :)

**Comment by disq** [comment]
> This looks good 🚀 Just missing a test for the new command :)

Added in [`980ac31` (#13941)](https://github.com/cloudquery/cloudquery/pull/13941/commits/980ac31eb2c6de72e8030553d87fcdf0b2a8be23)

File: cli/cmd/install_test.go [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		},
+		{
+			name:   "multiple_destinations",
+			config: "multiple-destinations.yml",
+			wantFiles: []string{
+				"cloudquery.log",
+				"plugins/destination/cloudquery/test/v2.1.0/plugin",
+				"plugins/destination/cloudquery/test/v2.1.0/plugin.zip",
+				"plugins/destination/cloudquery/test/v2.2.7/plugin",
+				"plugins/destination/cloudquery/test/v2.2.7/plugin.zip",
```
Comment (hermanschaaf) [inline cli/cmd/install_test.go:51]: I believe these versions get updated automatically by bots (https://github.com/cloudquery/cloudquery/blob/main/cli/cmd/testdata/multiple-destinations.yml#L19) so we will need to read them from the config or keep them in sync some other way for those automated PRs to succeed

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
Looks good, just one comment about testing

File: cli/cmd/install_test.go (hunk 3 above)
Comment (erezrokah) [reply cli/cmd/install_test.go:51]: This is the regex we use https://github.com/cloudquery/.github/blob/c43e6ce1c7e117aa0da14aff24ec15df002fe0a6/.github/renovate-default.json5#L87 so if you create a new spec without the comment it won't automatically update the versions

Comment (erezrokah) [reply cli/cmd/install_test.go:51]: For example https://github.com/cloudquery/cloudquery/blob/54cecdc91bb1af0501ac2ff4f65b4a4a448305bc/cli/cmd/testdata/multiple-destinations.yml#L6 pins the source version

**Review by erezrokah — APPROVED** [review]

**Pull request merged**